### PR TITLE
Fixing ActivitySource from DiagnosticSource

### LINKED ISSUE #1522: DiagnosticSource instrumentations produce activities without ActivitySource.
Activities created outside ActivitySource API (i.e DiagnosticSource APIs) has a default ActivitySource with name as string.Empty. ActivitySource is the instrumentationlibrary information, which would be missing for those activities.
There is no easy fix, other than to use reflection and set Activity.ActivitySource.

https://github.com/open-telemetry/opentelemetry-dotnet/issues/19#issuecomment-725772716

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1522

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -26,8 +26,11 @@ namespace OpenTelemetry.Instrumentation.AspNet.Implementation
 {
     internal class HttpInListener : ListenerHandler
     {
+        internal const string ActivitySourceName = "OpenTelemetry.Http";
```
Comment (eddynaka) [inline src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs:29]: should we differentiate webclient from httpclient?

**Comment by eddynaka** [comment]
Running benchmark, I could not see any difference:
**BEFORE**
|                        Method |     Mean |   Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------------------ |---------:|--------:|--------:|-------:|------:|------:|----------:|
| InstrumentedAspNetCoreGetPage | 229.5 us | 4.52 us | 9.33 us | 1.4648 |     - |     - |   2.21 KB |

**AFTER**
|                        Method |     Mean |   Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------------------ |---------:|--------:|--------:|-------:|------:|------:|----------:|
| InstrumentedAspNetCoreGetPage | 226.8 us | 4.18 us | 3.91 us | 1.4648 |     - |     - |   2.21 KB |

File: src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs (hunk 1 above)
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs:29]:
```suggestion
        internal static readonly string ActivitySourceName = typeof(HttpInListener).Assembly.GetName().Name;
```

**Comment by cijothomas** [comment]
image
Confirmed that this fixed the OTLPExporter issue :)

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -11,15 +11,16 @@
   `CustomProperty`.
   ([#1463](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1463))
 * Remove RentrantExportProcessor as it is not required by spec.
+* `ActivitySourceAdapter.Start` requires `ActivitySource` to update new
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:14]: please change to `ActivitySourceAdapter` supports setting ActivitySource for Activities created without ActivitySource.

Comment (cijothomas) [reply src/OpenTelemetry/CHANGELOG.md:14]: Also add changelog to every instrumentation - HttpInstrumentation sets ActivitySource to activities created outside ActivitySource etc.

**Review by cijothomas — APPROVED** [review]
LGTM

**Pull request merged**